In [ ]:
# Ubicacional ocal del archivo
import os
print(os.getcwd())

d:\FABRIZIO\Documents\UPCtemporal\2026-1\Prg-Cnc-Distr\PC2


# **GENERAR DATASET**
**Dataset: [INDECOPI]-[OPS1] Expedientes Presentados**
Contiene dos archivos CSV de los años 2015 y 2016.
- [INDECOPI]-[OPS1] Expedientes Presentados - 2015
- [INDECOPI]-[OPS1] Expedientes Presentados - 2016

El presente código tiene como objetivo generar datos de forma sintética para cumplir el objetivo de un millón de registros solicitados en el presente proyecto.
Se tomará en cuenta la relación entre ciertas variables y su distribución para mantener una partición de valores únicos similar al conjunto de datos original.

Link: https://www.datosabiertos.gob.pe/dataset/indecopi-ops1-expedientes-presentados


In [ ]:
import pandas as pd
# Cargar los dos csv
df1 = pd.read_csv("INDECOPI_OPS1_ExpedientesPresentados_2015.csv", sep=";")
df2 = pd.read_csv("INDECOPI_OPS1_ExpedientesPresentados_2016.csv", sep=";")

In [ ]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2363 entries, 0 to 2362
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VC_LUGAR              2363 non-null   object 
 1   VC_TIPO_EXPEDIENTE    2363 non-null   object 
 2   EMPLEADO              2358 non-null   object 
 3   FEC_PRE               2363 non-null   object 
 4   FEC_REING             853 non-null    object 
 5   NRO_EXPEDIENTE        2363 non-null   object 
 6   FEC_CAL               2363 non-null   object 
 7   SAC                   2363 non-null   int64  
 8   COD_C                 2363 non-null   object 
 9   VC_RUBRO              2219 non-null   object 
 10  DENUNCIANTES          2363 non-null   object 
 11  DENUNCIADOS           2363 non-null   object 
 12  RUC_DENUNCIADOS       2363 non-null   object 
 13  EXPEDIENTE_PRINCIPAL  779 non-null    object 
 14  AREA_CPC              0 non-null      float64
 15  EXP_ORIGEN_ORPS      

In [ ]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2779 entries, 0 to 2778
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VC_LUGAR              2779 non-null   object 
 1   VC_TIPO_EXPEDIENTE    2779 non-null   object 
 2   EMPLEADO              2779 non-null   object 
 3   FEC_PRE               2779 non-null   object 
 4   FEC_REING             1052 non-null   object 
 5   NRO_EXPEDIENTE        2779 non-null   object 
 6   FEC_CAL               2779 non-null   object 
 7   SAC                   2779 non-null   int64  
 8   COD_C                 2779 non-null   object 
 9   VC_RUBRO              2314 non-null   object 
 10  DENUNCIANTES          2777 non-null   object 
 11  DENUNCIADOS           2775 non-null   object 
 12  RUC_DENUNCIADOS       2775 non-null   object 
 13  EXPEDIENTE_PRINCIPAL  916 non-null    object 
 14  AREA_CPC              0 non-null      float64
 15  EXP_ORIGEN_ORPS      

El csv de 2015 presenta 2363 registros.

El csv de 2016 presenta 2779 registros.

In [ ]:
print(df1.columns.equals(df2.columns))  # Verificar que pueden unirse

True


In [ ]:
df = pd.concat([df1, df2], ignore_index=True)   # Unir dataframes

In [ ]:
df.info()   # Dataframe original - Registros entre 2015 a 2016

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5142 entries, 0 to 5141
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VC_LUGAR              5142 non-null   object 
 1   VC_TIPO_EXPEDIENTE    5142 non-null   object 
 2   EMPLEADO              5137 non-null   object 
 3   FEC_PRE               5142 non-null   object 
 4   FEC_REING             1905 non-null   object 
 5   NRO_EXPEDIENTE        5142 non-null   object 
 6   FEC_CAL               5142 non-null   object 
 7   SAC                   5142 non-null   int64  
 8   COD_C                 5142 non-null   object 
 9   VC_RUBRO              4533 non-null   object 
 10  DENUNCIANTES          5140 non-null   object 
 11  DENUNCIADOS           5138 non-null   object 
 12  RUC_DENUNCIADOS       5138 non-null   object 
 13  EXPEDIENTE_PRINCIPAL  1695 non-null   object 
 14  AREA_CPC              0 non-null      float64
 15  EXP_ORIGEN_ORPS      

Cantidad de registros iniciales: 5142

Columnas sin valores nulos:
- VC_LUGAR
- VC_TIPO_EXPEDIENTE
- FEC_PRE               
- NRO_EXPEDIENTE        
- FEC_CAL               
- SAC                 
- COD_C                   
- DESCR_CIIU 

In [ ]:
print("Cantidad de valores de VC_LUGAR: ", df.VC_LUGAR.unique().size)
print("Cantidad de valores de VC_TIPO_EXPEDIENTE: ", df.VC_TIPO_EXPEDIENTE.unique().size)
print("Cantidad de valores de EMPLEADO: ", df.EMPLEADO.unique().size)
print("Cantidad de valores de FEC_PRE: ", df.FEC_PRE.unique().size)
print("Cantidad de valores de NRO_EXPEDIENTE: ", df.NRO_EXPEDIENTE.unique().size)
print("Cantidad de valores de FEC_CAL: ", df.FEC_CAL.unique().size)
print("Cantidad de valores de SAC: ", df.SAC.unique().size)
print("Cantidad de valores de COD_C: ", df.COD_C.unique().size)
print("Cantidad de valores de DESCR_CIIU: ", df.DESCR_CIIU.unique().size)

Cantidad de valores de VC_LUGAR:  1
Cantidad de valores de VC_TIPO_EXPEDIENTE:  7
Cantidad de valores de EMPLEADO:  28
Cantidad de valores de FEC_PRE:  515
Cantidad de valores de NRO_EXPEDIENTE:  3091
Cantidad de valores de FEC_CAL:  695
Cantidad de valores de SAC:  2
Cantidad de valores de COD_C:  5
Cantidad de valores de DESCR_CIIU:  104


In [ ]:
# Revisar los casos donde NRO EXPEDIENTE se repite
df001 = df[df.duplicated(subset="NRO_EXPEDIENTE", keep=False)]
df001 = df001.sort_values(by="NRO_EXPEDIENTE")

In [ ]:
df001.head(20) # Revisar primeros 20 registros

,VC_LUGAR,VC_TIPO_EXPEDIENTE,EMPLEADO,FEC_PRE,FEC_REING,NRO_EXPEDIENTE,FEC_CAL,SAC,COD_C,VC_RUBRO,DENUNCIANTES,DENUNCIADOS,RUC_DENUNCIADOS,EXPEDIENTE_PRINCIPAL,AREA_CPC,EXP_ORIGEN_ORPS,DESCR_CIIU
2250,SEDE CENTRAL,INCUM.LIQUID.DE COSTAS Y COSTOS,ROJAS CONTRERAS JOSE ALBERTO,2015-01-09,2015-02-18,000001-2015/PS1,2015-04-01,0,C,SERVICIOS VARIOS,"BAZUALDO MELENDEZ, RODOLFO JOSE",SUPERMERCADOS PERUANOS SOCIEDAD ANONIMA 'O ' S...,20100070970,000589-2013/PS3,NaN,NaN,Venta al por menor en almacenes no especializa...
2065,SEDE CENTRAL,LIQUIDACION DE COSTAS Y COSTOS,ZAPANA CARO FLOR ESTELA,2015-01-05,NaN,000001-2015/PS1,2015-02-16,0,D,SISTEMA FINANCIERO BANCARIO,"ZEVALLOS MORALES, JOSE ANTONIO",BANCO RIPLEY PERU S.A.,20259702411,002236-2013/PS2,NaN,NaN,Otros tipos de intermediación monetaria
1524,SEDE CENTRAL,INCUMPL. DE ACUERDO CONCILIAT.,JIMENEZ CHAVEZ CATHERINE JOAHNA,2015-01-12,NaN,000001-2015/PS1,2015-02-23,0,C,"ACTIVIDADES ARTISTICAS, DE ENTRETENIMIENTO Y E...","FELIX ORE, AUGUSTO MARCOS",SUN INVERSIONES SOCIEDAD ANONIMA CERRADA,20486229684,000802-2014/CC2,NaN,NaN,Otras actividades de entretenimiento n.c.p.
1793,SEDE CENTRAL,INCUMPL. DE MEDIDAS CORRECTIVAS,ZAPANA CARO FLOR ESTELA,2015-01-08,2015-03-09,000001-2015/PS1,2015-04-22,0,C,SERVICIOS VARIOS,"GUERRA YUMPE, CARLOS JAVIER",PRENSA POPULAR S.A.C.,20502560574,001808-2012/CPC,NaN,NaN,Actividades de agencias de noticias
2355,SEDE CENTRAL,INCUMPL. DE ACUERDO CONCILIATORIO SAC,ZAPANA CARO FLOR ESTELA,2015-01-08,2015-04-17,000001-2015/PS1,2015-06-01,0,C,"ACTIVIDADES ARTISTICAS, DE ENTRETENIMIENTO Y E...","ESPINOZA SOSA, FIORELLA","CARDENAS TEXI, CARLOS ALBERTO",10107352711,NaN,NaN,NaN,Otras actividades de servicios n.c.p.
497,SEDE CENTRAL,DENUNCIA,JIMENEZ CHAVEZ CATHERINE JOAHNA,2015-01-05,NaN,000001-2015/PS1,2015-02-16,0,C,AGENCIAS DE VIAJE Y OTROS SERVICIOS DE TRANSPORTE,"ALEGRIA GOMEZ, CRISTIAM",DESPEGAR.COM PERU SAC,20544547756,NaN,NaN,NaN,Actividades de agencias de viajes y organizado...
5109,SEDE CENTRAL,INCUMPL. DE ACUERDO CONCILIATORIO SAC,MONDRAGON FERNANDEZ ELIZABETH,2016-01-06,NaN,000001-2016/PS1,2016-02-17,0,C,NaN,"PORTUGUEZ MAURTUA, DOMINGO MARCELO",INVERSIONES Y PROYECTOS EL ALAMO SOCIEDAD ANON...,20507880089,NaN,NaN,NaN,Otras actividades empresariales n.c.p.
5027,SEDE CENTRAL,INCUM.LIQUID.DE COSTAS Y COSTOS,PALOMINO QUINTO MILUSKA,2016-01-04,2016-07-07,000001-2016/PS1,2016-08-22,0,C,COMERCIO MINORISTA DE OTROS PRODUCTOS,"ORELLANA BALTTOLU DE TAM, ELVIRA","VARGAS JESUS, CESAR ALBERTO",45379839,001695-2013/PS3,NaN,NaN,Otros tipos de venta al por menor no realizada...
2790,SEDE CENTRAL,DENUNCIA,JARA CAMPOS MADELEINE,2016-01-04,NaN,000001-2016/PS1,2016-02-15,0,B,TELECOMUNICACIONES,"BECERRA BELLO, JACOBO",ENTEL PERU S.A.,20106897914,NaN,NaN,NaN,Telecomunicaciones
4139,SEDE CENTRAL,INCUMPL. DE ACUERDO CONCILIAT.,COLLAO VERA MARIA BELEN,2016-01-05,NaN,000001-2016/PS1,2016-02-16,0,C,CONSTRUCCION E INMOBILIARIOS,"ELORREAGA NARRO, JONNY FERNANDO",TOTAL INMUEBLES S.R.L.,20414851241,000292-2015/CC2,NaN,NaN,Construcción de edificios completos o de parte...


**Observaciones iniciales**

- VC_LUGAR solo posee un valor único: SEDE CENTRAL
- Existen tres fechas:
    - FEC_PRE   : Fecha de presentación del expediente (Nunca es nulo)
    - FEC_REING : Fecha de reingreso del expediente (Puede ser nulo, no siempre hay reingreso)
    - FEC_CAL   : Fecha de revisión del expediente (Nunca es nulo)
- La variable SAC es booleana, pero el dataset no explica a qué hace referencia.
- La variable COD_C es categórica, pero el dataset no explica a qué hace referencia o cómo clasifica los expedientes.
- Las variables AREA_CPC y EXP_ORIGEN_ORPS son siempre nulas.

In [ ]:
print("Cantidad de valores de VC_RUBRO: ", df.VC_RUBRO.unique().size)
print("Cantidad de valores de DESCR_CIIU: ", df.DESCR_CIIU.unique().size)

Cantidad de valores de VC_RUBRO:  37
Cantidad de valores de DESCR_CIIU:  104


- Existen varios DESCR_CIIU para un mismo VC_RUBRO.

In [ ]:
print("Cantidad de valores de DENUNCIANTES: ", df.DENUNCIANTES.unique().size)
print("Cantidad de valores de DENUNCIADOS: ", df.DENUNCIADOS.unique().size)
print("Cantidad de valores de RUC_DENUNCIADOS: ", df.RUC_DENUNCIADOS.unique().size)

Cantidad de valores de DENUNCIANTES:  4125
Cantidad de valores de DENUNCIADOS:  1551
Cantidad de valores de RUC_DENUNCIADOS:  1383


- La cantidad de valores unicos de DENUNCIADOS no es igual a RUC_DENUNCIADOS

**Comentarios del análisis preliminar**

- VC_LUGAR no es importante, siempre es el mismo valor. Es preferible eliminar el campo.
- VC_TIPO_EXPEDIENTE existen 7 tipos de expedientes. Mantener variable.
- EMPLEADO es el nombre del empleado que ha supervisador el caso. Hay 28 empleados. Hay pocos registros nulos que pueden eliminarse o imputar.
- FEC_PRE es fecha de presentación. Mantener variable.
- FEC_REING es fecha de reingreso del expediente. Mantener variable.
- NRO_EXPEDIENTE tiene 3091 valores unicos. Indica que hay expedientes con el mismo numero. Por lo visto, el numero puede repetirse para cada tipo de VC_TIPO_EXPEDIENTE. Es decir, puede existir el mismo expediente numero "000001-2015/PS1" para "DENUNCIA" y para "LIQUIDACION DE COSTAS Y COSTOS", etc.
- FEC_CAL es fecha de revisión del reclamo por INDECOPI. Mantener variable.
- SAC es una variable binaria. No se menciona qué representa.  Mantener variable.
- COD_C es una variable de categoría. Divide los reclamos en 5 categorías: [A, B, C, D, E].  No se menciona qué representa.  Mantener variable.
- VC_RUBRO está muy ligada a DESCR_CIIU, ya que existen varios DESCR_CIIU para un solo VC_RUBRO. Cuando se genere datos, se debe considerar generar primero un Rubro y luego asignar un DESCR_CIIU en base al rubro.
- DENUNCIANTES es el nombre del denunciante. Se generará nombres aleatorios con posibilidad de repetir el mismo denunciante.
- DENUNCIADOS y RUC_DENUNCIADOS deben ir de la mano. Se generará denunciados repetidos.
- EXPEDIENTE_PRINCIPAL indica si el expediente proviene de otro expediente en OPS o CC.
- AREA_CPC es un campo nulo. No hay datos. Eliminar.
- EXP_ORIGEN_ORPS es un campo nulo. No hay datos. Eliminar.
- DESCR_CIIU depende de VC_RUBRO.

In [ ]:
df.to_csv("INDECOPI_OPS1.csv", index=False, sep=";", encoding="utf-8-sig") # descargar CSV actual

___

Generación de datos nuevos

In [ ]:
import pandas as pd
# recargar el CSV guardado
dfY = pd.read_csv("INDECOPI_OPS1.csv", sep=";")

In [ ]:
dfY.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5142 entries, 0 to 5141
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VC_LUGAR              5142 non-null   object 
 1   VC_TIPO_EXPEDIENTE    5142 non-null   object 
 2   EMPLEADO              5137 non-null   object 
 3   FEC_PRE               5142 non-null   object 
 4   FEC_REING             1905 non-null   object 
 5   NRO_EXPEDIENTE        5142 non-null   object 
 6   FEC_CAL               5142 non-null   object 
 7   SAC                   5142 non-null   int64  
 8   COD_C                 5142 non-null   object 
 9   VC_RUBRO              4533 non-null   object 
 10  DENUNCIANTES          5140 non-null   object 
 11  DENUNCIADOS           5138 non-null   object 
 12  RUC_DENUNCIADOS       5138 non-null   object 
 13  EXPEDIENTE_PRINCIPAL  1695 non-null   object 
 14  AREA_CPC              0 non-null      float64
 15  EXP_ORIGEN_ORPS      

In [ ]:
dfY["VC_LUGAR"].unique()

array(['SEDE CENTRAL'], dtype=object)

**GENERAR VARIABLE - VC_LUGAR**

In [ ]:
# Crear nuevo dataset dfX
n = 1000000

# Generar dataset final dfX
# Primera columna - unico valor - SEDE CENTRAL
dfX = pd.DataFrame({
    "VC_LUGAR": ["SEDE CENTRAL"] * n
})

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 1 columns):
 #   Column    Non-Null Count    Dtype 
---  ------    --------------    ----- 
 0   VC_LUGAR  1000000 non-null  object
dtypes: object(1)
memory usage: 7.6+ MB


**GENERAR VARIABLE - VC_TIPO_EXPEDIENTE**
- Se extrae la probabilidad de aparición de cada tipo de expediente en base al dataset original.
- Se generan datos mantenimiento el ratio original.

In [ ]:
# Para seleccionar los tipos de expediente / sacar probabilidad-ratio de aparicion

dist_tipoExp = dfY["VC_TIPO_EXPEDIENTE"].value_counts(normalize=True)
valores_tipoExp = dist_tipoExp.index
probs_tipoExp = dist_tipoExp.values

print("Tipos de expediente: ", valores_tipoExp)
print("Probabilidades: ", probs_tipoExp)

Tipos de expediente:  Index(['DENUNCIA', 'LIQUIDACION DE COSTAS Y COSTOS',
       'INCUMPL. DE MEDIDAS CORRECTIVAS', 'INCUM.LIQUID.DE COSTAS Y COSTOS',
       'INCUMPL. DE ACUERDO CONCILIATORIO SAC',
       'INCUMPL. DE ACUERDO CONCILIAT.', 'SANCIONADOR POR EL ART. 5'],
      dtype='object', name='VC_TIPO_EXPEDIENTE')
Probabilidades:  [6.35744846e-01 1.43329444e-01 1.19603267e-01 4.76468300e-02
 3.44224037e-02 1.90587320e-02 1.94476857e-04]


In [ ]:
# Generar datos de la columna VC_TIPO_EXPEDIENTE
import numpy as np

dfX["VC_TIPO_EXPEDIENTE"] = np.random.choice(
    valores_tipoExp,
    size=len(dfX),
    p=probs_tipoExp
)

dfX.info()
dfX["VC_TIPO_EXPEDIENTE"].unique()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 2 columns):
 #   Column              Non-Null Count    Dtype 
---  ------              --------------    ----- 
 0   VC_LUGAR            1000000 non-null  object
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object
dtypes: object(2)
memory usage: 15.3+ MB


array(['DENUNCIA', 'LIQUIDACION DE COSTAS Y COSTOS',
       'INCUMPL. DE ACUERDO CONCILIATORIO SAC',
       'INCUMPL. DE MEDIDAS CORRECTIVAS',
       'INCUM.LIQUID.DE COSTAS Y COSTOS',
       'INCUMPL. DE ACUERDO CONCILIAT.', 'SANCIONADOR POR EL ART. 5'],
      dtype=object)

In [ ]:
# Revisar resultados
dist_tipoExp = dfX["VC_TIPO_EXPEDIENTE"].value_counts(normalize=True)
print("Tipos de expediente: ", dist_tipoExp.index)
print("Probabilidades: ", dist_tipoExp.values)

Tipos de expediente:  Index(['DENUNCIA', 'LIQUIDACION DE COSTAS Y COSTOS',
       'INCUMPL. DE MEDIDAS CORRECTIVAS', 'INCUM.LIQUID.DE COSTAS Y COSTOS',
       'INCUMPL. DE ACUERDO CONCILIATORIO SAC',
       'INCUMPL. DE ACUERDO CONCILIAT.', 'SANCIONADOR POR EL ART. 5'],
      dtype='object', name='VC_TIPO_EXPEDIENTE')
Probabilidades:  [6.36341e-01 1.43597e-01 1.19043e-01 4.73710e-02 3.44130e-02 1.90400e-02
 1.95000e-04]


Mantiene una distribución similar a la original.

**GENERAR VARIABLE - EMPLEADO**

In [ ]:
# Realizar mismo proceso con EMPLEADO

dist_empl = dfY["EMPLEADO"].value_counts(normalize=True)
valores_empl = dist_empl.index
probs_empl = dist_empl.values

dfX["EMPLEADO"] = np.random.choice(
    valores_empl,
    size=len(dfX),
    p=probs_empl
)

dfX.info()
print(dfX["EMPLEADO"].unique().size)
dfX["EMPLEADO"].unique()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 3 columns):
 #   Column              Non-Null Count    Dtype 
---  ------              --------------    ----- 
 0   VC_LUGAR            1000000 non-null  object
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object
 2   EMPLEADO            1000000 non-null  object
dtypes: object(3)
memory usage: 22.9+ MB
27


array(['ROCA ORTEGA PAMELA', 'MEZA CALLE LUIS ELIAS',
       'MONDRAGON FERNANDEZ ELIZABETH', 'QUIÑONES MARTINEZ GISELLE',
       'CACERES GUERRA CARLOS ANDRES', 'HURTADO ABARCA SANDRA',
       'FALCON GIRON GIOVANNA', 'ALPACA FEBRES DANIEL ERNESTO',
       'MONZON ALVAREZ CLAUDIA DEL PILAR', 'PALOMINO QUINTO MILUSKA',
       'ZAPANA CARO FLOR ESTELA', 'QUIROZ LEON CLAUDIA ESMERALDA',
       'COLLAO VERA MARIA BELEN', 'SACHA CASTILLO LESLY ANDRIELA',
       'SALVATIERRA CORDOVA KATHERINE', 'JARA CAMPOS MADELEINE',
       'CARRANZA OLIVO MAURICIO JAVIER', 'ROJAS CONTRERAS JOSE ALBERTO',
       'JIMENEZ CHAVEZ CATHERINE JOAHNA', 'MUNIVE CASTRO PATRICIA SUMMER',
       'CHAVEZ GUTIERREZ WENDY ELIZABETH', 'ZAGAL MALAGA PATRICIA',
       'ROSALES VICENTE KESHIA MELANY', 'FERNANDEZ DELGADO WALTER',
       'LARREA CUEVA KARLA', 'LEZCANO VASQUEZ MAYRA',
       'ASCOY BALCAZAR GUILLERMO IVAN'], dtype=object)

**GENERAR VARIABLE - FEC_PRE**
- Se genera fechas random en años posteriores a 2016.

In [ ]:
# Generar fechas random

# rango de fechas
inico = np.datetime64('2017-01-01')
fin = np.datetime64('2025-06-30')

# generar fechas aleatorias
fechas = np.random.randint(inico.astype('int64'), fin.astype('int64'), n)

# convertir a datetime
dfX["FEC_PRE"] = pd.to_datetime(fechas)

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 5 columns):
 #   Column              Non-Null Count    Dtype         
---  ------              --------------    -----         
 0   VC_LUGAR            1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object        
 2   EMPLEADO            1000000 non-null  object        
 3   FEC_PRE             1000000 non-null  datetime64[ns]
 4   FEC_REING           371040 non-null   datetime64[ns]
dtypes: datetime64[ns](2), object(3)
memory usage: 38.1+ MB


**GENERAR VARIABLE - FEC_REING**
- Considerando que puede ser nulo. Se buscar mantener un ratio nulo-fecha similar al dataset original.
- La fecha generada debe ser posterior a FEC_PRE.

In [ ]:
# FEC_REING se calcula considerando una probabilidad similar a la original
prop_reing = dfY["FEC_REING"].notna().mean()

# Mantener distribucion de dias entre presentacion y reingreso
dfY_temp = dfY.dropna(subset=["FEC_PRE", "FEC_REING"]).copy()

dfY_temp["dias"] = (pd.to_datetime(dfY_temp["FEC_REING"]) - 
                    pd.to_datetime(dfY_temp["FEC_PRE"])).dt.days

dias_reing = dfY_temp["dias"]
dias_reing = dias_reing[dias_reing >= 0] # solo valores validos

# Decidir los registros no nulos
mascara_reing = np.random.rand(n) < prop_reing
# Elegir dias random
dias_muestra = np.random.choice(dias_reing, size=n)
# Sumar dias random
fec_reing = dfX["FEC_PRE"] + pd.to_timedelta(dias_muestra, unit="D")
# Limitar fecha
max_fecha = pd.to_datetime("2025-12-31")
fec_reing = fec_reing.where(fec_reing <= max_fecha, max_fecha)
# Crear la columna "FEC_REING" considerando los nulos
dfX["FEC_REING"] = pd.NaT
dfX.loc[mascara_reing, "FEC_REING"] = fec_reing[mascara_reing]

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 5 columns):
 #   Column              Non-Null Count    Dtype         
---  ------              --------------    -----         
 0   VC_LUGAR            1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object        
 2   EMPLEADO            1000000 non-null  object        
 3   FEC_PRE             1000000 non-null  datetime64[ns]
 4   FEC_REING           371028 non-null   datetime64[ns]
dtypes: datetime64[ns](2), object(3)
memory usage: 38.1+ MB


**GENERAR VARIABLE - NRO_EXPEDIENTE**
- Es un código identificador que aumenta de 1 en 1.
- El orden depende de la fecha de presentación.

In [ ]:
# Generar numero de expediente NRO_EXPEDIENTE

# ordenar por fecha
dfX = dfX.sort_values("FEC_PRE").reset_index(drop=True)

# Extraer-armar datos
dfX["ANIO"] =dfX["FEC_PRE"].dt.year
dfX["NUMCORR"] = dfX.groupby("ANIO").cumcount()+1
dfX["NUMCORR"] = dfX["NUMCORR"].astype(str).str.zfill(6) # llenar con 0

# Construir NRO_EXPEDIENTE
dfX["NRO_EXPEDIENTE"] = (dfX["NUMCORR"] + "-" + dfX["ANIO"].astype(str) + "/PS1")

# Eliminar columnas temporales
dfX = dfX.drop(columns=["ANIO","NUMCORR"])

dfX.info()
dfX.head(4)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 6 columns):
 #   Column              Non-Null Count    Dtype         
---  ------              --------------    -----         
 0   VC_LUGAR            1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object        
 2   EMPLEADO            1000000 non-null  object        
 3   FEC_PRE             1000000 non-null  datetime64[ns]
 4   FEC_REING           371028 non-null   datetime64[ns]
 5   NRO_EXPEDIENTE      1000000 non-null  object        
dtypes: datetime64[ns](2), object(4)
memory usage: 45.8+ MB


,VC_LUGAR,VC_TIPO_EXPEDIENTE,EMPLEADO,FEC_PRE,FEC_REING,NRO_EXPEDIENTE
0,SEDE CENTRAL,INCUM.LIQUID.DE COSTAS Y COSTOS,ROCA ORTEGA PAMELA,1970-01-01 00:00:00.000017167,NaT,000001-1970/PS1
1,SEDE CENTRAL,DENUNCIA,JIMENEZ CHAVEZ CATHERINE JOAHNA,1970-01-01 00:00:00.000017167,1970-01-21 00:00:00.000017167,000002-1970/PS1
2,SEDE CENTRAL,INCUMPL. DE MEDIDAS CORRECTIVAS,QUIROZ LEON CLAUDIA ESMERALDA,1970-01-01 00:00:00.000017167,NaT,000003-1970/PS1
3,SEDE CENTRAL,INCUMPL. DE ACUERDO CONCILIATORIO SAC,MEZA CALLE LUIS ELIAS,1970-01-01 00:00:00.000017167,NaT,000004-1970/PS1


___

Recordatorio de las variables

In [ ]:
dfY.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5142 entries, 0 to 5141
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   VC_LUGAR              5142 non-null   object 
 1   VC_TIPO_EXPEDIENTE    5142 non-null   object 
 2   EMPLEADO              5137 non-null   object 
 3   FEC_PRE               5142 non-null   object 
 4   FEC_REING             1905 non-null   object 
 5   NRO_EXPEDIENTE        5142 non-null   object 
 6   FEC_CAL               5142 non-null   object 
 7   SAC                   5142 non-null   int64  
 8   COD_C                 5142 non-null   object 
 9   VC_RUBRO              4533 non-null   object 
 10  DENUNCIANTES          5140 non-null   object 
 11  DENUNCIADOS           5138 non-null   object 
 12  RUC_DENUNCIADOS       5138 non-null   object 
 13  EXPEDIENTE_PRINCIPAL  1695 non-null   object 
 14  AREA_CPC              0 non-null      float64
 15  EXP_ORIGEN_ORPS      

___

**GENERAR VARIABLE - FEC_CAL**
- No es nulo.
- Sigue un proceso similar a FEC_REING. Es posterior a FEC_PRE.

In [ ]:
# Generar FEC_CAL - Mismo proceso que FEC_REING pero sin probabilidades
dfY_temp2 = dfY.dropna(subset=["FEC_PRE", "FEC_CAL"]).copy()

# Rangos de dias entre PRE y CAL
dfY_temp2["dias_cal"] = (
    pd.to_datetime(dfY_temp2["FEC_CAL"]) - 
    pd.to_datetime(dfY_temp2["FEC_PRE"])
).dt.days

# solo valores válidos
diasCAL = dfY_temp2["dias_cal"]
diasCAL = diasCAL[diasCAL >= 0]

# generar dias
diasCAL_muestra = np.random.choice(diasCAL, size=n)
# generar fechas CAL
fec_cal = dfX["FEC_PRE"] + pd.to_timedelta(diasCAL_muestra, unit="D")
# limitar a fin de año 2025
fec_cal = fec_cal.where(fec_cal <= max_fecha, max_fecha)
# Asignar al dataset
dfX["FEC_CAL"] = fec_cal

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 7 columns):
 #   Column              Non-Null Count    Dtype         
---  ------              --------------    -----         
 0   VC_LUGAR            1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object        
 2   EMPLEADO            1000000 non-null  object        
 3   FEC_PRE             1000000 non-null  datetime64[ns]
 4   FEC_REING           371028 non-null   datetime64[ns]
 5   NRO_EXPEDIENTE      1000000 non-null  object        
 6   FEC_CAL             1000000 non-null  datetime64[ns]
dtypes: datetime64[ns](3), object(4)
memory usage: 53.4+ MB


**GENERAR VARIABLE - SAC**
- Se mantiene una distribución similar al dataset original.

In [ ]:
# Generar columna SAC

# Probabilidad para mantener distribucion
prop_SAC = dfY["SAC"].mean()

# Generar
dfX["SAC"] = np.random.choice(
    [0, 1],
    size=n,
    p=[1 - prop_SAC, prop_SAC]
)

**GENERAR VARIABLE - COD_C**
- Se desconocer representación.
- Se decidió ignorar.

In [ ]:
# Generar columna COD_C                 

# Desconozo si existe alguna relacion con otras variables.
# De forma temporal se asigna un valor "Z" temporal.
dfX["COD_C"] = "Z"

**GENERAR VARIABLE - VC_RUBRO**
- Se mantiene una distribución similar al dataset original.
- Se generan valores nulos para posterior limpieza.

In [ ]:
# Generar VC_RUBRO - Mantener distribucion
dist_RUBRO = dfY["VC_RUBRO"].value_counts(normalize=True, dropna=False)
valores_RUBRO = dist_RUBRO.index
probs_RUBRO = dist_RUBRO.values

# Generar datos - se considera los nulos
dfX["VC_RUBRO"] = np.random.choice(
    valores_RUBRO,
    size=n,
    p=probs_RUBRO
)

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 10 columns):
 #   Column              Non-Null Count    Dtype         
---  ------              --------------    -----         
 0   VC_LUGAR            1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object        
 2   EMPLEADO            1000000 non-null  object        
 3   FEC_PRE             1000000 non-null  datetime64[ns]
 4   FEC_REING           371028 non-null   datetime64[ns]
 5   NRO_EXPEDIENTE      1000000 non-null  object        
 6   FEC_CAL             1000000 non-null  datetime64[ns]
 7   SAC                 1000000 non-null  int32         
 8   COD_C               1000000 non-null  object        
 9   VC_RUBRO            881383 non-null   object        
dtypes: datetime64[ns](3), int32(1), object(6)
memory usage: 72.5+ MB


**GENERAR VARIABLE - DENUNCIANTES**
- Se genera nombres de denunciantes basados en nombres y apellidos comunes en Perú.
- Cantidad máxima de nombres únicos:
    - 100 casos de un solo nombre
    - 100x100 = 10000 casos de dos nombres
    - 50x50 = 2500 combinaciones de apellidos
    - Resultado: (100+10000)x2500 = 25250000 combinaciones de nombres completos posibles

In [ ]:
# Generar DENUNCIANTES

np.random.seed(42) # reproducir

apellidos = np.array([
    "QUISPE","HUAMAN","MAMANI","FLORES","GARCIA","RODRIGUEZ","PEREZ",
    "LOPEZ","SANCHEZ","RAMIREZ","TORRES","VARGAS","CASTILLO","ROJAS",
    "DIAZ","ALVAREZ","ROMERO","NAVARRO","CHAVEZ","SALAZAR","VASQUEZ",
    "RIVERA","CASTRO","ESPINOZA","HERRERA","GONZALES","RAMOS","HURTADO",
    "MORALES","CRUZ","REYES","ORTIZ","SOTO","SILVA","MENDOZA","PAREDES",
    "AGUILAR","PEÑA","VALDEZ","DELGADO","CAMPOS","GUERRERO","FUENTES",
    "CARRILLO","MIRANDA","VEGA","CARRASCO","SANTOS","PALOMINO","LEON"
])
nombres = np.array([
    "JUAN","CARLOS","LUIS","JOSE","MIGUEL","JORGE","PEDRO","DANIEL","FERNANDO","RICARDO",
    "RAUL","ALBERTO","CESAR","EDUARDO","OSCAR","MANUEL","HUGO","VICTOR","FRANCISCO","DAVID",
    "DIEGO","ANDRES","JAVIER","SERGIO","ROBERTO","MARTIN","ARMANDO","ENRIQUE","LUCIANO","GUSTAVO",
    "ANGEL","IVAN","BRUNO","ADRIAN","SAMUEL","LEONARDO","PABLO","EMILIO","CRISTIAN","FELIPE",
    "ANTONIO","ALONSO","SEBASTIAN","ALEX","MAXIMO","RENZO","BRYAN","JHON","WILMER","EDWIN",
    "MARIA","ANA","LUISA","CARMEN","ROSA","ELENA","PATRICIA","SANDRA","GLORIA","MARTA",
    "TERESA","ISABEL","BEATRIZ","SUSANA","VERONICA","LUCIA","ANDREA","PAOLA","CARLA","DIANA",
    "MONICA","CECILIA","NATALIA","CLAUDIA","KAREN","LILIANA","YOLANDA","SONIA","ROCIO","ESTHER",
    "FABIOLA","MELISSA","JESSICA","DANIELA","VALERIA","ALEJANDRA","CAMILA","XIMENA","MARIBEL","NOEMI",
    "ANGELA","SOFIA","RENATA","BRIGITTE","DAYANA","MILAGROS","YESENIA","KATHERINE","PAMELA","CINTHIA"
])

# Generar listas de nombres
apellido1 = np.random.choice(apellidos, size=n)
apellido2 = np.random.choice(apellidos, size=n)
nombre1 = np.random.choice(nombres, size=n)
nombre2 = np.random.choice(nombres, size=n)

# Usar series de panda
apellido1_s = pd.Series(apellido1)
apellido2_s = pd.Series(apellido2)
nombre1_s = pd.Series(nombre1)
nombre2_s = pd.Series(nombre2)

# 70% tendrán segundo nombre
usar_segundo = np.random.rand(n) < 0.7

# combinar nombres
nombres_final = np.where(
    usar_segundo,
    nombre1_s + " " + nombre2_s,
    nombre1_s
)

# Guardar DENUNCIANTE creados
dfX["DENUNCIANTES"] = (apellido1_s + " " + apellido2_s + ", " + nombres_final)

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 11 columns):
 #   Column              Non-Null Count    Dtype         
---  ------              --------------    -----         
 0   VC_LUGAR            1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE  1000000 non-null  object        
 2   EMPLEADO            1000000 non-null  object        
 3   FEC_PRE             1000000 non-null  datetime64[ns]
 4   FEC_REING           371028 non-null   datetime64[ns]
 5   NRO_EXPEDIENTE      1000000 non-null  object        
 6   FEC_CAL             1000000 non-null  datetime64[ns]
 7   SAC                 1000000 non-null  int32         
 8   COD_C               1000000 non-null  object        
 9   VC_RUBRO            881383 non-null   object        
 10  DENUNCIANTES        1000000 non-null  object        
dtypes: datetime64[ns](3), int32(1), object(7)
memory usage: 80.1+ MB


In [ ]:
dfX["DENUNCIANTES"].unique().size

865237

Existe un millon de registros. Sin embargo, existen 865237 nombres completos únicos.

Se concluye que existen nombres repetidos lo cual está bien ya que es lo que buscamos.

**GENERAR VARIABLE - DENUNCIADOS y RUC_DENUNCIADOS**
- Se generan repeticiones de denunciados, considerando que el RUC y el denunciado forman parte de una misma entidad.

In [ ]:
# DENUNCIADOS y RUC_DENUNCIADOS
# se va a mantener la distribucion original para identificar empresas con varios expedientes
dfY["PAR_DENUNCIADO"] = (
    dfY["DENUNCIADOS"].astype(str) + ";" +
    dfY["RUC_DENUNCIADOS"].astype(str)
) # unir en una columna
dist_PAR = dfY["PAR_DENUNCIADO"].value_counts(normalize=True)
valores_PAR = dist_PAR.index
probs_PAR = dist_PAR.values

# generar datos
PAR_muestra = np.random.choice(valores_PAR, size=n, p=probs_PAR)
# Separar en columnas
df_tempPAR = pd.Series(PAR_muestra).str.split(";", expand=True)
# Guardar datos en dataset
dfX["DENUNCIADOS"] = df_tempPAR[0]
dfX["RUC_DENUNCIADOS"] = df_tempPAR[1]

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 17 columns):
 #   Column                Non-Null Count    Dtype         
---  ------                --------------    -----         
 0   VC_LUGAR              1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE    1000000 non-null  object        
 2   EMPLEADO              1000000 non-null  object        
 3   FEC_PRE               1000000 non-null  datetime64[ns]
 4   FEC_REING             371028 non-null   datetime64[ns]
 5   NRO_EXPEDIENTE        1000000 non-null  object        
 6   FEC_CAL               1000000 non-null  datetime64[ns]
 7   SAC                   1000000 non-null  int32         
 8   COD_C                 1000000 non-null  object        
 9   VC_RUBRO              881383 non-null   object        
 10  DENUNCIANTES          1000000 non-null  object        
 11  RUC_DENUNCIADOS       1000000 non-null  object        
 12  DENUNCIADOS           1000000 non-null  obj

**GENERAR VARIABLE - EXPEDIENTE_PRINCIPAL, AREA_CPC y EXP_ORIGEN_ORPS**
- Valores nulos.
- Se dicidió ignorar EXPEDIENTE_PRINCIPAL.

In [ ]:
# Columnas con valores nulos
dfX["EXPEDIENTE_PRINCIPAL"] = np.nan
dfX["AREA_CPC"] = np.nan
dfX["EXP_ORIGEN_ORPS"] = np.nan

dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 17 columns):
 #   Column                Non-Null Count    Dtype         
---  ------                --------------    -----         
 0   VC_LUGAR              1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE    1000000 non-null  object        
 2   EMPLEADO              1000000 non-null  object        
 3   FEC_PRE               1000000 non-null  datetime64[ns]
 4   FEC_REING             371028 non-null   datetime64[ns]
 5   NRO_EXPEDIENTE        1000000 non-null  object        
 6   FEC_CAL               1000000 non-null  datetime64[ns]
 7   SAC                   1000000 non-null  int32         
 8   COD_C                 1000000 non-null  object        
 9   VC_RUBRO              881383 non-null   object        
 10  DENUNCIANTES          1000000 non-null  object        
 11  RUC_DENUNCIADOS       1000000 non-null  object        
 12  DENUNCIADOS           1000000 non-null  obj

**GENERAR VARIABLE - DESCR_CIIU**
- Depende de la variable VC_RUBRO.
- Un VC_RUBRO puede tener multiples DESCR_CIIU.
- Se mantiene una distribución en DESCR_CIIU en base a VC_RUBRO similar ala original.

In [ ]:
from collections import defaultdict

# DESCR_CIIU es un campo dependiente de VC_RUBRO.

df_temp3 = dfY.dropna(subset=["VC_RUBRO", "DESCR_CIIU"]).copy()
dist_condicional = (
    df_temp3
    .groupby("VC_RUBRO")["DESCR_CIIU"]
    .value_counts(normalize=True)
) # Existen probabilidade distitnas de DESCR_CIIU para cada VC_RUBRO

# Usar diccionario para facilidad de acceso
dic_dist = defaultdict(lambda: ([], []))

for (rubro, ciiu), prob in dist_condicional.items():
    dic_dist[rubro][0].append(ciiu) # valor CIIU
    dic_dist[rubro][1].append(prob) # probabilida del valor

def obtener_ciiu(rubro):
    if rubro in dic_dist:
        valores, probs = dic_dist[rubro]
        return np.random.choice(valores, p=probs)
    else:
        return np.nan   # no existe el rubro
    
# Generar el campo
dfX["DESCR_CIIU"] = dfX["VC_RUBRO"].apply(obtener_ciiu)

In [ ]:
dfX.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 17 columns):
 #   Column                Non-Null Count    Dtype         
---  ------                --------------    -----         
 0   VC_LUGAR              1000000 non-null  object        
 1   VC_TIPO_EXPEDIENTE    1000000 non-null  object        
 2   EMPLEADO              1000000 non-null  object        
 3   FEC_PRE               1000000 non-null  datetime64[ns]
 4   FEC_REING             371028 non-null   datetime64[ns]
 5   NRO_EXPEDIENTE        1000000 non-null  object        
 6   FEC_CAL               1000000 non-null  datetime64[ns]
 7   SAC                   1000000 non-null  int32         
 8   COD_C                 1000000 non-null  object        
 9   VC_RUBRO              881383 non-null   object        
 10  DENUNCIANTES          1000000 non-null  object        
 11  RUC_DENUNCIADOS       1000000 non-null  object        
 12  DENUNCIADOS           1000000 non-null  obj

**RESULTADO**

In [ ]:
dfX.head(10)

,VC_LUGAR,VC_TIPO_EXPEDIENTE,EMPLEADO,FEC_PRE,FEC_REING,NRO_EXPEDIENTE,FEC_CAL,SAC,COD_C,VC_RUBRO,DENUNCIANTES,RUC_DENUNCIADOS,DENUNCIADOS,EXPEDIENTE_PRINCIPAL,AREA_CPC,EXP_ORIGEN_ORPS,DESCR_CIIU
0,SEDE CENTRAL,INCUM.LIQUID.DE COSTAS Y COSTOS,ROCA ORTEGA PAMELA,1970-01-01 00:00:00.000017167,NaT,000001-1970/PS1,1971-03-29 00:00:00.000017167,0,Z,TRANSPORTE POR VIA AEREA,"VALDEZ DELGADO, GLORIA BRIGITTE",20467534026,AMERICA MOVIL PERU S.A.C.,NaN,NaN,NaN,Transporte regular por vía aérea
1,SEDE CENTRAL,DENUNCIA,JIMENEZ CHAVEZ CATHERINE JOAHNA,1970-01-01 00:00:00.000017167,1970-01-21 00:00:00.000017167,000002-1970/PS1,1970-02-12 00:00:00.000017167,0,Z,CONSTRUCCION E INMOBILIARIOS,"MORALES DELGADO, OSCAR CARMEN",20467534026,AMERICA MOVIL PERU S.A.C.,NaN,NaN,NaN,Actividades inmobiliarias realizadas con biene...
2,SEDE CENTRAL,INCUMPL. DE MEDIDAS CORRECTIVAS,QUIROZ LEON CLAUDIA ESMERALDA,1970-01-01 00:00:00.000017167,NaT,000003-1970/PS1,1970-04-11 00:00:00.000017167,0,Z,TRANSPORTE TERRESTRE Y OTROS TIPOS DE TRANSPORTE,"DIAZ RAMOS, LUCIA VERONICA",20100070970,SUPERMERCADOS PERUANOS SOCIEDAD ANONIMA 'O ' S...,NaN,NaN,NaN,Otros tipos de transporte regular de pasajeros...
3,SEDE CENTRAL,INCUMPL. DE ACUERDO CONCILIATORIO SAC,MEZA CALLE LUIS ELIAS,1970-01-01 00:00:00.000017167,NaT,000004-1970/PS1,1970-02-15 00:00:00.000017167,0,Z,FABRICACION DE MUEBLES,"FUENTES ALVAREZ, KATHERINE VALERIA",20100041953,RIMAC SEGUROS Y REASEGUROS,NaN,NaN,NaN,Fabricación de muebles
4,SEDE CENTRAL,DENUNCIA,PALOMINO QUINTO MILUSKA,1970-01-01 00:00:00.000017167,NaT,000005-1970/PS1,1970-02-13 00:00:00.000017167,0,Z,SEGUROS Y PENSIONES,"LOPEZ MORALES, DANIEL MANUEL",20341841357,LAN PERU S.A.,NaN,NaN,NaN,Planes de seguros generales
5,SEDE CENTRAL,DENUNCIA,FALCON GIRON GIOVANNA,1970-01-01 00:00:00.000017167,NaT,000006-1970/PS1,1971-09-30 00:00:00.000017167,1,Z,SERVICIOS VARIOS,"VASQUEZ SANCHEZ, CARMEN",20100105862,BANCO FINANCIERO DEL PERU S.A.,NaN,NaN,NaN,Venta al por menor en almacenes no especializa...
6,SEDE CENTRAL,INCUM.LIQUID.DE COSTAS Y COSTOS,PALOMINO QUINTO MILUSKA,1970-01-01 00:00:00.000017167,NaT,000007-1970/PS1,1971-11-29 00:00:00.000017167,1,Z,NaN,"VALDEZ AGUILAR, ALEX NATALIA",20100043140,SCOTIABANK PERU S.A.A.,NaN,NaN,NaN,NaN
7,SEDE CENTRAL,DENUNCIA,PALOMINO QUINTO MILUSKA,1970-01-01 00:00:00.000017167,1970-08-09 00:00:00.000017167,000008-1970/PS1,1970-02-12 00:00:00.000017167,0,Z,SEGUROS Y PENSIONES,"CHAVEZ DIAZ, GLORIA",20100177774,TELEFONICA MOVILES S.A.,NaN,NaN,NaN,Actividades auxiliares de la financiación de p...
8,SEDE CENTRAL,INCUMPL. DE MEDIDAS CORRECTIVAS,JARA CAMPOS MADELEINE,1970-01-01 00:00:00.000017167,NaT,000009-1970/PS1,1970-04-09 00:00:00.000017167,0,Z,SISTEMA FINANCIERO BANCARIO,"CASTRO MORALES, DANIEL GUSTAVO",20100041953,RIMAC SEGUROS Y REASEGUROS,NaN,NaN,NaN,Otros tipos de intermediación monetaria
9,SEDE CENTRAL,LIQUIDACION DE COSTAS Y COSTOS,MONDRAGON FERNANDEZ ELIZABETH,1970-01-01 00:00:00.000017167,1970-02-01 00:00:00.000017167,000010-1970/PS1,1970-02-13 00:00:00.000017167,0,Z,NaN,"TORRES SOTO, DIEGO LUCIA",20106897914,ENTEL PERU S.A.,NaN,NaN,NaN,NaN


In [ ]:
# Descargar el dataset
dfX.to_csv("INDECOPI_OPS1_Millon.csv", index=False, sep=";", encoding="utf-8-sig", date_format="%Y-%m-%d")

Consideraciones:
- COD_C - ignorado.
- EXPEDIENTE_PRINCIPAL - ignorado.
